gamemode/core/hooks/sv_hooks.lua

```diff
@@ -118,10 +118,10 @@ function GM:CanPlayerInteractItem(client, action, item)
 end
 
 function GM:CanPlayerTakeItem(client, item)
-	if (type(item) == "Entity") then
+	if (type(item.entity) == "Entity") then
 		local char = client:getChar()
 
-		if (item.nutSteamID and item.nutSteamID == client:SteamID() and item.nutCharID != char:getID()) then
+		if (item.entity.nutSteamID and item.entity.nutSteamID == client:SteamID() and item.entity.nutCharID != char:getID()) then
```

Comment:
To prevent lines from being really long, you can break up the `if` statement into many lines like:
```lua
if (
  someLongCondition and
  someOtherLongCondition and
) then
  -- stuff
end
```

Reply:
You could shorten the check to `item.entity.nutSteamID == client:SteamID()` since `nil` is never equal to a steam ID anyways.